Library/Homebrew/extend/pathname.rb

```diff
@@ -436,11 +436,17 @@ def env_script_all_files(dst, env)
   end
 
   # Writes an exec script that invokes a java jar
-  def write_jar_script(target_jar, script_name, java_opts = "")
+  def write_jar_script(target_jar, script_name, java_opts = "", java_version: nil)
     mkpath
+    java_home = if java_version
+      <<~EOS
+        export JAVA_HOME="$(#{Language::Java.java_home_cmd(java_version)})"
```

Comment:
Is it worth skipping the `export` and just putting the var on the same line?

Reply:
Sure. Amended.